### ISSUE TITLE
MD051: Add support for custom heading anchor IDs

### ISSUE THREAD

**Issue by scottamain** [DavidAnson/markdownlint#830]
Many MD parsers allow for custom anchor names on headings as follows:

```
## Some Section {#name}
```

With this markup, I can change the heading but keep the anchor name the same and avoid breaking links.

The problem is that MD051 will now flag this as an invalid link:

```
See [some section](#name)
```

It would be great if markdownlint could capture `{#name}` as the heading ID to avoid this false positive.

(Although some MD parsers also allow you to use this syntax to add IDs on other elements, I believe headings are the most common and would be a great starting point to add this support. Then if there's demand, perhaps add support for custom IDs on other elements such as images or other arbitrary elements.)

**Comment by DavidAnson** [comment]
https://github.com/DavidAnson/markdownlint/blob/main/doc/ReleaseProcess.md
